library/platform.c

```diff
@@ -125,9 +125,9 @@ int mbedtls_platform_win32_vsnprintf( char *s, size_t n, const char *fmt, va_lis
         return( -1 );
 
 #if defined(_TRUNCATE)
-    ret = _vsnprintf_s( s, n, _TRUNCATE, fmt, arg );
+    ret = vsnprintf_s( s, n, _TRUNCATE, fmt, arg );
 #else
-    ret = _vsnprintf( s, n, fmt, arg );
+    ret = vsnprintf( s, n, fmt, arg );
```

Comment:
Purpose of ```mbedtls_platform_win32_vsnprintf``` was to provide C99 complaint function. Now if we have ```vsnprintf``` on Windows then we don't require this function all together. 
Since @mpg originally added this function. I will ask for his opinion.

Reply:
Do you mean to say that the statement [here](https://docs.microsoft.com/en-gb/cpp/c-runtime-library/reference/vsnprintf-vsnprintf-vsnprintf-l-vsnwprintf-vsnwprintf-l) is not correct. Snippet:
```
Beginning with the UCRT in Visual Studio 2015 and Windows 10, vsnprintf is no longer identical to _vsnprintf. The vsnprintf function complies with the C99 standard; _vnsprintf is retained for backward compatibility with older Visual Studio code.
```
If we are not sure about Windows behaviour recently. We can write some tests to confirm C99 compliance. We also need to maintain uniform behaviour across new and old versions of Windows and VS.

Reply:
No, I'm not saying this statement is incorrect. I guess the important part is "last time I checked" which apparently was prior to Windows 10. That's excellent news, I wasn't aware of this, good that MS improved their C99 support in that area, thanks for letting me know!

Unfortunately, unless we explicitly support only VS2015+Windows10 and above, I'm afraid we can't fully enjoy the comfort of a C99-compliant vsnprint() just yet, can we?

I fully agree regarding tests. Currently we have `test_snprintf()` in `tests/suites/main_test.function`. We probably want something similar for `vsnprintf()` just to be sure.

Reply:
So essentially we can change ```_WIN32``` to something that precisely tells when ```vsnprintf``` became C99 compliant on Windows.

Reply:
Yes, if we can detect that with an `#if` that would be great: if we have compliant versions of `vsnprintf` and `snprintf`, just use them directly, and otherwise, include the old code for building our own compliant versions and registering them as `mbedtls_(v)snprintf`.

Reply:
So, in the `development` branch we're aiming to support Windows 7 and up, and in the LTS branches, `mbedtls-2.7` and `mbedtls-2.1` we're supporting the older versions of Windows.

We need to only invoke `vsnprintf()` if the version of Windows is correct. I'd suggest putting some guards around it using `_WIN32_WINNT` to check the version.

cc: @k-stachowiak